boot/bootutil/src/image_validate.c

```diff
@@ -605,311 +379,84 @@ bootutil_img_validate(struct enc_key_data *enc_state, int image_index,
-                break;
-            }
-
-            if (type == IMAGE_TLV_DECOMP_SHA) {
-                /* Verify the image hash. This must always be present. */
-                if (len != sizeof(hash)) {
-                    rc = -1;
-                    goto out;
-                }
-                rc = LOAD_IMAGE_DATA(hdr, fap, off, buf, sizeof(hash));
-                if (rc) {
-                    goto out;
-                }
-
-                FIH_CALL(boot_fih_memequal, fih_rc, hash, buf, sizeof(hash));
-                if (FIH_NOT_EQ(fih_rc, FIH_SUCCESS)) {
-                    FIH_SET(fih_rc, FIH_FAILURE);
-                    goto out;
-                }
-
-                image_hash_valid = 1;
-            }
-        }
-
-        rc = !image_hash_valid;
-        if (rc) {
-            goto out;
-        }
-
-#ifdef EXPECTED_SIG_TLV
-#ifdef EXPECTED_KEY_TLV
-        rc = bootutil_tlv_iter_begin(&it, hdr, fap, EXPECTED_KEY_TLV, false);
-        if (rc) {
-            goto out;
-        }
-
-        if (it.tlv_end > bootutil_max_image_size(fap)) {
-            rc = -1;
-            goto out;
-        }
-
-        while (true) {
-            rc = bootutil_tlv_iter_next(&it, &off, &len, &type);
-            if (rc < 0) {
-                goto out;
-            } else if (rc > 0) {
-                break;
-            }
-
-            if (type == EXPECTED_KEY_TLV) {
-                /*
-                 * Determine which key we should be checking.
-                 */
-                if (len > KEY_BUF_SIZE) {
-                    rc = -1;
-                    goto out;
-                 }
-#ifndef MCUBOOT_HW_KEY
-                rc = LOAD_IMAGE_DATA(hdr, fap, off, buf, len);
-                if (rc) {
-                    goto out;
-                }
-                key_id = bootutil_find_key(buf, len);
-#else
-                rc = LOAD_IMAGE_DATA(hdr, fap, off, key_buf, len);
-                if (rc) {
-                    goto out;
-                }
-                key_id = bootutil_find_key(image_index, key_buf, len);
-#endif /* !MCUBOOT_HW_KEY */
-                /*
-                 * The key may not be found, which is acceptable.  There
-                 * can be multiple signatures, each preceded by a key.
-                 */
-            }
-        }
-#endif /* EXPECTED_KEY_TLV */
-
-        rc = bootutil_tlv_iter_begin(&it, hdr, fap, IMAGE_TLV_DECOMP_SIGNATURE, true);
-        if (rc) {
-            goto out;
-        }
-
-        if (it.tlv_end > bootutil_max_image_size(fap)) {
-            rc = -1;
-            goto out;
-        }
-
-        while (true) {
-            rc = bootutil_tlv_iter_next(&it, &off, &len, &type);
-            if (rc < 0) {
-                goto out;
-            } else if (rc > 0) {
-                rc = 0;
-                break;
-            }
+    if (0 != ed25519_sign_open(hash, sizeof(hash), pub,
+                            *(const ed25519_signature *)sig)){
```

Comment:
Maybe anti-glitch with `ensure()` or doubling the call to better protect against glitch (nRF is very glitch-prone, at least revisions before 3.x were?

Reply:
added a FIH macro invocation in https://github.com/trezor/mcuboot/commit/06eb505517f0886282f89660024faf519f9b6aa0

which will call the funciton twice and compare in two ways, we just need to set correct CONFIG_BOOT_FIH_PROFILE_HIGH or (MEDIUM) in mcuboot config in main repo